src/herder/TransactionQueue.cpp

```diff
@@ -193,14 +293,27 @@ TransactionQueue::shift()
     auto it = std::begin(mPendingTransactions);
     while (it != end)
     {
-        if (mPendingDepth == ++(it->second.mAge))
+        if (!it->second.mTransactions.empty())
```

Comment:
why the conditional increment?
* Either this can "never happen" - because `extract` is supposed to delete empty transaction in which case, why test for it
* if it can happen, this would cause `mPendingTransactions` to grow unbounded (right now, the increment acts as an automatic "memory leak cleanup" for those queues that are left behind)

either case seems to support always incrementing